### PR TITLE
fix: skip empty metadata filter in RetrieverTool to prevent vector store error

### LINKED ISSUE #4900: Retriever tool will throw error if "additional metadata filter" is added then removed
### Describe the bug

Retriever tool will throw error if "additional metadata filter" is added then removed.

Error:
(error:"You must enter a `filter` object with at least one key-value pair.")

The only way I found to resolve is to replace the retriever tool node completely.

### To Reproduce

1. From Retriever Tool, add a filter from Additional Properties > "additional metadata filter"
2. Close modal and save
3. Remove the filter in "additional metadata filter" added in step 1.
4. When the bot uses the tool, error is returned: You must enter a `filter` object with at least one key-value pair

### Expected behavior

Removing the filter should not result in error.

### Screenshots

_No response_

### Flow

_No response_

### Use Method

None

### Flowise Version

_No response_

### Operating System

None

### Browser

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by majiayu000**
## Summary

Fixes #4900

When a user adds an "Additional Metadata Filter" in the Retriever Tool node then removes it, the empty filter object `{}` was still passed to the vector store. This caused Pinecone (and potentially other vector stores) to reject the query with:

> "You must enter a `filter` object with at least one key-value pair."

The fix adds a guard to check that the resolved metadata filter has at least one key-value pair before assigning it to `vectorStore.filter`. This follows the same pattern already used in `Pinecone_LlamaIndex.ts`.

## Changes

- `packages/components/nodes/tools/RetrieverTool/RetrieverTool.ts`: Add `Object.keys().length > 0` check before setting `vectorStore.filter`
- `packages/components/nodes/tools/RetrieverTool/RetrieverTool.test.ts`: Add 6 tests covering empty filter, non-empty filter, null filter, and end-to-end string-to-object parsing scenarios

## Test plan

- [x] New unit tests pass (6 test cases)
- [x] Existing tests unaffected
- [x] Verified empty `{}` filter is not passed to vector store
- [x] Verified non-empty filter `{"category": "docs"}` is correctly applied
- [x] Verified null/undefined filter is handled safely

### CODE REVIEW

File: packages/components/nodes/tools/RetrieverTool/RetrieverTool.test.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        it('should not set filter when metadata filter is a string "{}" parsed to empty object', () => {
+            const mockVectorStore = { filter: undefined }
+            const retrieverToolMetadataFilter = '{}'
+
+            if (retrieverToolMetadataFilter) {
+                const metadatafilter =
+                    typeof retrieverToolMetadataFilter === 'object'
+                        ? retrieverToolMetadataFilter
+                        : JSON.parse(retrieverToolMetadataFilter)
+                const newMetadataFilter = resolveFlowObjValue(metadatafilter, {})
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply packages/components/nodes/tools/RetrieverTool/RetrieverTool.test.ts]: Fixed. Renamed `metadatafilter` to `metadataFilter` in both the test file and the source file for camelCase consistency.

**Comment by majiayu000** [comment]
Thanks for the suggestion about `Pinecone.ts`! You're right that lines 254-258 have the same pattern. However, the Pinecone case is slightly different — the filter is used as part of `PineconeStoreParams` during store initialization (not at query time), and there's follow-up logic on lines 259-266 that may append `$or` conditions to `obj.filter`. Adding an empty-check guard there could affect that downstream logic.

I think it's better to address that in a separate PR with its own testing to avoid unintended side effects. I'll open a follow-up issue for it.
